### DISCUSSION TITLE
Visual State Manager - Material TextField

### DISCUSSION THREAD

**Discussion by seanjaeger73** [enisn/UraniumUI#462] [Q&A] [answered]
I am attempting to change the border thickness on visual state Focused for the Material TextField input.  This works as an inline style and even a contentview resource placed in the resourcedictionary... but if I place this in the "Styles.xaml" as key'd or unkey'd, the application crashes with the following error:

```
[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: Microsoft.Maui.Controls.Xaml.XamlParseException: Position 12:37. Type converter failed: Exception has been thrown by the target of an invocation.
[mono-rt]  ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
[mono-rt]  ---> Microsoft.Maui.Controls.Xaml.XamlParseException: Position 76:37. The PropertyName of TextField.BorderThicknessProperty is not BorderThickness
```

This is the setter I am trying to use:

```xml
    <Style TargetType="material:TextField">
        <Setter Property="VisualStateManager.VisualStateGroups">
            <VisualStateGroupList x:Name="CommonStates">
                <VisualStateGroup>
                    <VisualState x:Name="Normal">
                        <VisualState.Setters>
                            <Setter Property="BorderThickness" Value="1" />
                        </VisualState.Setters>
                    </VisualState>
                    <VisualState x:Name="Focused">
                        <VisualState.Setters>
                            <Setter Property="BorderThickness" Value="2" />
                        </VisualState.Setters>
                    </VisualState>
                </VisualStateGroup>
            </VisualStateGroupList>
        </Setter>
    </Style>
```
Any help would be appreciated.

**Comment by enisn** [comment, marked as answer]
You're right, because the name of the BorderThicknessProperty is defined as `BorderColor` by mistake. I'll create an issue and fix it ina patch version

https://github.com/enisn/UraniumUI/blob/cc982f9e435626f258632060af9d30a0ab41e643/src/UraniumUI.Material/Controls/InputField.cs#L382C16-L382C27

**Comment by seanjaeger73** [comment]
I was finally able to implement the border thickness after the fix applied.  Unfortunately, any thickness set greater than 1.2 or so messes up the floating label:

![image](https://github.com/enisn/UraniumUI/assets/122306912/5f1008c7-4544-4cb9-9033-65a60296d4a2)

![image](https://github.com/enisn/UraniumUI/assets/122306912/755d1746-dd61-4f09-bdce-0cabc2b714de)

Is there another style that I can set to increase left padding on the floating label?

Thank you,
Sean

**Reply by enisn** [reply]
Hi, calculations doesn't include border thickness parameter, I'll add border thickness parameter to the algorithm.


Until then, as a workaround you can target to **Label**s inside TextField with [Cascading Styling](https://enisn-projects.io/docs/en/uranium/latest/theming/CascadingStyling) and set Padding for it:
```xml
        <Style TargetType="material:TextField">
            <Setter Property="uranium:CascadingStyle.Resources">
                <ResourceDictionary>
                    <Style TargetType="Label">
                        <Setter Property="Padding" Value="20,0,0,0" />
                    </Style>
                </ResourceDictionary>
            </Setter>
        </Style>
```

**Reply by seanjaeger73** [reply]
Thank you @enisn - I was able to get the border thickness to update on focus as shown in the screenshots, but now there is a large gap to the right of the label that varies textbox to textbox:

![image](https://github.com/enisn/UraniumUI/assets/122306912/d7e269bb-1dad-4467-ac1a-83ef48f366d3)

![image](https://github.com/enisn/UraniumUI/assets/122306912/592a8440-3233-4140-9b17-bffc3288a6de)

I tried a few different styles but could not find one that made any difference.

Thank you,
Seab

**Comment by enisn** [comment]
Resolved by https://github.com/enisn/UraniumUI/pull/574

It'll be released in the next v2.8 release soon
